plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -497,6 +497,11 @@ CacheStats getStatisticsCacheStats()
         cache.invalidateAll(cacheKeys);
     }
 
+    public void onDataChanged(JdbcTableHandle handle)
```

Comment:
`JdbcTableHandle` is not a good representation of the table. For example, we don't want to distinguish between "a plan table" and "table with selected columns", or "a table with a constraint" here.

Reply:
https://github.com/trinodb/trino/pull/8237

Reply:
That was the point. To allow invalidate only partial information about the table.